tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolLookupTest.java

```diff
@@ -358,6 +358,34 @@ public void testSymbolLookupInFields(int line, int column, int expSymbols, List<
         };
     }
 
+    @Test(dataProvider = "OnFailSymbolPosProvider")
+    public void testOnFailClauseSymbolLookup(int line, int column, int expSymbols, List<String> expSymbolNames) {
+        Project project = BCompileUtil.loadProject("test-src/on_fail_symbol_lookup_test.bal");
+        Package currentPackage = project.currentPackage();
+        ModuleId defaultModuleId = currentPackage.getDefaultModule().moduleId();
+        PackageCompilation packageCompilation = currentPackage.getCompilation();
+        SemanticModel model = packageCompilation.getSemanticModel(defaultModuleId);
+        Document srcFile = getDocumentForSingleSource(project);
+
+        BLangPackage pkg = packageCompilation.defaultModuleBLangPackage();
+        ModuleID moduleID = new BallerinaModuleID(pkg.packageID);
+
+        Map<String, Symbol> symbolsInFile = getSymbolsInFile(model, srcFile, line, column, moduleID);
+        assertEquals(symbolsInFile.size(), expSymbols);
+
+        for (String symName : expSymbolNames) {
+            assertTrue(symbolsInFile.containsKey(symName), "Symbol not found: " + symName);
+        }
+    }
+
+    @DataProvider(name = "OnFailSymbolPosProvider")
+    public Object[][] getOnFailSymbolPositions() {
+        List<String> expSymbolNames = List.of("err", "test", "v", "errRef");
+        return new Object[][]{
+                {25, 23, 4, expSymbolNames}
+        };
+    }
```

Comment:
Are you planning on adding any more test cases to this? If not, no need to use a data provider.

Reply:
Updated in b82a084b8bb603675a2afe7a0d16ca65a086b214